src/mainwindow/mainwindow.cpp

```diff
@@ -1662,6 +1669,33 @@ bool MainWindow::slot_exportBaseMap()
     return saveFile(fileNames[0], SaveModeEnum::BASEMAP, SaveFormatEnum::MM2);
 }
 
+bool MainWindow::slot_exportMm2xmlMap()
+{
+    const auto makeSaveDialog = [this]() {
+        // FIXME: code duplication
+        auto save = std::make_unique<QFileDialog>(this,
+                                                  "Choose map file name ...",
+                                                  QDir::current().absolutePath());
+        save->setFileMode(QFileDialog::AnyFile);
+        save->setDirectory(QDir(getConfig().autoLoad.lastMapDirectory));
+        save->setNameFilter("MM2XML Maps (*.mm2xml)");
+        save->setDefaultSuffix("mm2xml");
+        save->setAcceptMode(QFileDialog::AcceptSave);
+        save->selectFile(QFileInfo(m_mapData->getFileName())
+                             .fileName()
+                             .replace(QRegularExpression(R"(\.mm2$)"), ".mm2xml"));
```

Comment:
This might not work in the future when we load a .mm2xml file and then try to export that same file again. We might need to fix this bug (and code duplication) in other parts of the code too.

Reply:
In my idea, `m_mapData->getFileName()` would always be a `.mm2` file, never a `.mm2xml` file.
From another perspective, loading `.mm2xml` file in my opinion should **not** set `m_mapData->getFileName()`

Does it make sense?

Reply:
Let me rephrase my idea: when loading a `.mm2xml` map, the map name or filename is **not** set in MMapper2.

After all, it's an import: users are supposed to save it back in `.mm2` format, and **that** will set the map name and filename.